samples/apply-dictionary-renderer-to-graphics-overlay/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertographicsoverlay/components/ApplyDictionaryRendererToGraphicsOverlayViewModel.kt

```diff
@@ -0,0 +1,235 @@
+    val graphicsOverlay: GraphicsOverlay = GraphicsOverlay()
+
+    // SceneViewProxy to enable programmatic viewpoint changes
+    val sceneViewProxy = SceneViewProxy()
+
+    // Used to display error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    // Provision path where downloaded sample assets will be placed by the downloader activity
+    private val provisionPath: String by lazy {
+        app.getExternalFilesDir(null)?.path.toString() + File.separator + app.getString(R.string.apply_dictionary_renderer_to_graphics_overlay_app_name)
+    }
+
+    init {
+        viewModelScope.launch {
+            // Load the scene first
+            arcGISScene.load().onFailure { throwable ->
+                messageDialogVM.showMessageDialog(throwable)
+            }
+
+            // Create and apply dictionary renderer from a web style
+            val dictionaryRenderer = createMil2525dDictionaryRenderer().getOrElse {
+                messageDialogVM.showMessageDialog(it)
+                return@launch
+            }
+
+            // Create the point graphics in separate coroutine from a local XML file
+            val pointGraphics = makeMessageGraphics().getOrElse {
+                messageDialogVM.showMessageDialog(it)
+                return@launch
+            }
+
+
+            // Set the graphics overlay to use the dictionary renderer and add graphics
+            graphicsOverlay.apply {
+                renderer = dictionaryRenderer
+                graphics.addAll(pointGraphics)
+            }
+
+            // Sets the camera to look a the graphics in the graphics overlay
+            graphicsOverlay.extent?.let { extent ->
+                sceneViewProxy.setViewpointCamera(
+                    camera = Camera(
+                        lookAtPoint = extent.center,
+                        distance = 15000.0,
+                        heading = 0.0,
+                        pitch = 70.0,
+                        roll = 0.0
+                    )
+                )
+            }
+        }
+    }
+
+    /**
+     * Create and load a [DictionarySymbolStyle] from a web style and use it to create a [DictionaryRenderer].
+     */
+    private suspend fun createMil2525dDictionaryRenderer(): Result<DictionaryRenderer> {
+        // Creates a dictionary symbol style from a dictionary style portal item.
+        val portalItem = PortalItem(
+            portal = Portal.arcGISOnline(Portal.Connection.Anonymous),
+            itemId = "d815f3bdf6e6452bb8fd153b654c94ca"
+        )
+
+        val dictionarySymbolStyle = DictionarySymbolStyle(portalItem = portalItem)
+
+        return dictionarySymbolStyle.load().mapCatching {
+            // Uses the "Ordered Anchor Points" for the symbol style draw rule.
+            dictionarySymbolStyle.configurations.firstOrNull { it.name.equals("model", ignoreCase = true) }
+                ?.let { configuration -> configuration.value = "ORDERED ANCHOR POINTS" }
```

Comment:
This is fairly dense code for a sample, and could use some step by step comments. How about 

```suggestion
// Get the model configuration from the style's list of configurations.
val modelConfiguration = dictionarySymbolStyle.configurations.firstOrNull { 
    it.name.equals("model", ignoreCase = true) 
}
if (modelConfigutation != null) {
    // Set the draw rule of the style to "Ordered Anchor Points" by setting the value of the 
    // configuration to "ORDERED ANCHOR POINTS".
    modelConfiguration.value = "ORDERED ANCHOR POINTS"
}




```